Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Rating.hs

```diff
@@ -56,24 +57,30 @@ handler merchantId req ride = do
       logTagInfo "FeedbackAPI" $
         "Updating existing rating for " +|| ride.id ||+ " with new rating " +|| ratingValue ||+ "."
       QRating.updateRating rideRating.id driverId ratingValue feedbackDetails
-  calculateAverageRating driverId merchant.minimumDriverRatesCount
+  calculateAverageRating driverId merchant.minimumDriverRatesCount ratingValue person.totalRatings person.totalRatingScore
 
 calculateAverageRating ::
   (EsqDBFlow m r, EncFlow m r) =>
   Id DP.Person ->
   Int ->
+  Int ->
+  Int ->
+  Int ->
   m ()
-calculateAverageRating personId minimumDriverRatesCount = do
+calculateAverageRating personId minimumDriverRatesCount ratingValue totalRatings totalRatingScore = do
   logTagInfo "PersonAPI" $ "Recalculating average rating for driver " +|| personId ||+ ""
-  allRatings <- B.runInReplica $ QRating.findAllRatingsForPerson personId
-  let ratingsSum = fromIntegral $ sum (allRatings <&> (.ratingValue))
-  let ratingCount = length allRatings
-  when (ratingCount == 0) $
+  let newRatingsCount = totalRatings + 1
+  let newTotalRatingScore = totalRatingScore + ratingValue
+  when (totalRatings == 0) $
     logTagInfo "PersonAPI" "No rating found to calculate"
-  when (ratingCount >= minimumDriverRatesCount) $ do
-    let newAverage = ratingsSum / fromIntegral ratingCount
-    logTagInfo "PersonAPI" $ "New average rating for person " +|| personId ||+ " , rating is " +|| newAverage ||+ ""
-    void $ QP.updateAverageRating personId newAverage
+  if totalRatings >= minimumDriverRatesCount
+    then do
+      let isValidRating = True
```

Comment:
No need of creating these variables. Can pass the boolean directly

Reply:
hey @0utkarsh , @ritikaJus has asked for more readability of code so lets keep them?

Reply:
I don't think it affects the readability in any way. If you still want it that way, you can add comments instead of extra variables